tests/test_main.py

```diff
@@ -44,3 +45,41 @@ def test_main_scan(self, mock_cfg, mock_env, mock_scanner):
         environment_instance.setup.assert_called_once()
         environment_instance.scanner.assert_called_once()
         mock_scanner.scan.assert_called_once()
+
+    @patch("py_sonar_scanner.scanner.Scanner")
+    @patch("py_sonar_scanner.__main__.Environment")
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_main_scan_fail(self, mock_sys, mock_env, mock_scanner):
+        mock_sys.argv = ["path/to/scanner/py-sonar-scanner"]
+        environment_instance = MagicMock()
+        environment_instance.setup = Mock(side_effect=Exception("Something"))
+        mock_scanner.scan = Mock()
+        environment_instance.scanner.return_value = mock_scanner
+        mock_env.return_value = environment_instance
+
+        with self.assertLogs(ApplicationLogger.get_logger()) as log:
+            scan()
+            self.assertEqual(2, len(log.records))
+            self.assertEqual("Error during SonarScanner execution: Something", log.records[0].getMessage())
+            self.assertFalse(log.records[0].exc_info)
+            self.assertEqual(
+                "Re-run SonarScanner using the -X switch to enable full debug logging.", log.records[1].getMessage()
+            )
+
+    @patch("py_sonar_scanner.scanner.Scanner")
+    @patch("py_sonar_scanner.__main__.Environment")
+    @patch("py_sonar_scanner.configuration.sys")
+    def test_main_scan_debug_fail(self, mock_sys, mock_env, mock_scanner):
+        mock_sys.argv = ["path/to/scanner/py-sonar-scanner", "-X"]
+
+        environment_instance = MagicMock()
+        environment_instance.setup = Mock(side_effect=Exception("Something"))
+        mock_scanner.scan = Mock()
+        environment_instance.scanner.return_value = mock_scanner
+        mock_env.return_value = environment_instance
+
+        with self.assertLogs(ApplicationLogger.get_logger()) as log:
+            scan()
+            self.assertEqual(1, len(log.records))
+            self.assertEqual("Error during SonarScanner execution: Something", log.records[0].getMessage())
```

Comment:
Maybe it would be nice to have an assertion on the log level since it might not be obvious to everyone that `exception` logs on `ERROR` level.